packages/venia-concept/src/components/Navigation/categoryTree.js

```diff
@@ -21,45 +24,81 @@ class Tree extends Component {
         ),
         onNavigate: func,
         rootNodeId: number.isRequired,
-        updateRootNodeId: func.isRequired
+        updateRootNodeId: func.isRequired,
+        currentId: number.isRequired
     };
 
     get leaves() {
-        const { nodes, onNavigate, rootNodeId, updateRootNodeId } = this.props;
-        const { childrenData: childNodeIds } = nodes[rootNodeId];
+        const {
+            classes,
+            onNavigate,
+            rootNodeId,
+            updateRootNodeId,
+            currentId
+        } = this.props;
 
-        const leaves = childNodeIds.reduce((elements, id) => {
-            const childNode = nodes[id];
-            const { childrenData, position } = childNode;
-            const isLeaf = !childrenData.length;
-            const elementProps = { nodeId: id, nodes };
+        return rootNodeId ? (
+            <Query query={navigationMenu} variables={{ id: rootNodeId }}>
+                {({ loading, error, data }) => {
+                    if (error) return <div>Data Fetch Error</div>;
+                    if (loading) return null;
 
-            const element = isLeaf ? (
-                <Leaf {...elementProps} onNavigate={onNavigate} />
-            ) : (
-                <Branch {...elementProps} onDive={updateRootNodeId} />
-            );
+                    const branches = [];
 
-            elements[position - 1] = <li key={id}>{element}</li>;
+                    const children = data.category.children.sort((a, b) => {
```

Comment:
I can't wait until Magento GraphQL ensures that ordering is significant, so we don't have to do this.